src/driver.rs

```diff
@@ -98,6 +108,42 @@ impl Driver {
         Ok(unsafe { Dataset::from_c_dataset(c_dataset) })
```

Comment:
The old function could call the new one instead of having two similar implementations.

Reply:
I agree.
I came up with this, which works, but I'm not sure passing empty key value pairs is a good idea. Is there a better way?

```rust
    pub fn create_with_band_type<T: GdalType>(
        &self,
        filename: &str,
        size_x: isize,
        size_y: isize,
        bands: isize,
    ) -> Result<Dataset> {
        let options = [RasterCreationOption { key: "", value: "" }];
        self.create_with_band_type_with_options::<T>(filename, size_x, size_y, bands, &options)
    }
```

Reply:
Oh, so it didn't work with `[]`?

I agree that the API is clunky, but that's a larger change and I can't promise anything about it.

Reply:
oh, yes. It does, sorry! :)